app/utils/build-url-with-prefix-map.js

```diff
@@ -0,0 +1,39 @@
+import Ember from 'ember';
+
+export default function buildURLWithPrefixMap(prefixPropertyMapping){
+  return function buildURL(type, id, record) {
+    var url = [],
+        host = Ember.get(this, 'host'),
+        prefix = this.urlPrefix();
```

Comment:
Sure. The no-mutiple-value-setting-var style is newish in Ember, but I've found it really useful. ember-data lags on best practices as usual :sob: